backend/ops_api/ops/document/document_gateway.py

```diff
@@ -9,35 +10,39 @@
 
 class DocumentGateway:
     def __init__(self, config: Config) -> None:
-        self.provider = config["AUTHLIB_OAUTH_CLIENTS"]
+        self.providers = config.get("AUTHLIB_OAUTH_CLIENTS", {})
         self.repository_factory = DocumentRepositoryFactory()
 
         # Validate and register providers with the factory
-        self.register_provider(self.provider)
+        self.register_providers(self.providers)
 
-    def register_provider(self, provider: str) -> None:
+        # Select provider based on the current user
+        if current_user.id >= 500:  # Users with id 5xx are test users
```

Comment:
As per our demo, please replace this with using the `config` object (file based config)